backend/src/apiserver/main.go

```diff
@@ -143,7 +144,11 @@ func startHttpProxy(resourceManager *resource.ResourceManager) {
 		io.WriteString(w, `{"commit_sha":"`+common.GetStringConfigWithDefault("COMMIT_SHA", "unknown")+`", "tag_name":"`+common.GetStringConfigWithDefault("TAG_NAME", "unknown")+`"}`)
 	})
 
-	topMux.Handle("/apis/", mux)
+	// log streaming is provided via HTTP.
+	runLogServer := server.NewRunLogServer(resourceManager)
+	topMux.HandleFunc("/apis/v1beta1/runs/{run_id}/nodes/{node_id}/log", runLogServer.ReadRunLog)
```

Comment:
Please use the existing convention of writing *.proto definitions for the endpoint and generate clients/server stubs for the endpoint. The benefits:
* KFP API SDK is auto generated
* KFP API Reference is auto generated

Feel free to let me know if you cannot find related resources to use the existing pattern

Reply:
@Bobgy Thanks for your comment and suggestions.
Before I switch to gRPC and http proxy I'd like to ask about support for realtime log streaming from k8s POD, which is quite important for us.

1. At first k8s logs API is based on HTTP chunked transfer encoding and the same is proposed here. k8s logs are by nature unstructured (only timestamp can optionally be extracted).
2. Secondly, the priority for me was the raw http endpoint that can be easily used (even with `curl`) to download logs in the original form.

My understanding is that any solution based on proto would add complexity to the message structure and downloaded log would require post-processing.

I would love to see a good and efficient pattern that addresses all requirements.
Thanks!

Reply:
These are great points! Totally understood your decision now.
Sounds good to me.